### PR TITLE
Release

### PR DESCRIPTION
**Pull request by osahyoun**

### CODE REVIEW

File: app/workers/queue_listener.rb [hunk 1]
```diff
@@ -28,12 +28,32 @@ def update_pages(params)
     data = params[:params]
     data.delete(:name)
 
-    client.update_petition_page( data.merge(id: petition_id(params) ))
-    client.update_donation_page( data.merge(id: donation_id(params) ))
+    client.update_petition_page(
+      data.merge({
+        id: petition_id(params)
+      }.tap do |load|
+          load[:title] = suffix_title(data[:title], 'petition') if data[:title]
```
Comment (NealJMD) [inline app/workers/queue_listener.rb]: do we ever submit a page without a title?

File: app/workers/queue_listener.rb [hunk 2]
```diff
@@ -28,12 +28,32 @@ def update_pages(params)
...
+        load[:title] = suffix_title(data[:title], 'donation') if data[:title]
+      end
+      )
+    )
   end
 
   private
 
+  def suffix_title(title, type)
+    return title if title.strip =~ %r{(#{type.capitalize})}
```
Comment (NealJMD) [inline app/workers/queue_listener.rb]: curious why you used `%r{}` and not just double quotes

File: spec/requests/pages_spec.rb [hunk 3]
```diff
@@ -21,17 +21,62 @@
...
+          VCR.use_cassette('page_update_200'){ post("/message", params) }
+        end
+
+        context 'title' do
+          it 'adds title suffix to petition page' do
+            expect_any_instance_of( ActionKitConnector::Client ).to(
+              receive(:update_petition_page).with(hash_including({title: "house of cards (Petition)"}))
+            )
+           expect_any_instance_of( ActionKitConnector::Client ).to(
+              receive(:update_donation_page).with(hash_including({title: "house of cards (Donation)"}))
```
Comment (NealJMD) [inline spec/requests/pages_spec.rb]: this spec makes sense to me :+1:

File: spec/requests/pages_spec.rb [hunk 4]
```diff
@@ -40,12 +85,12 @@
...
-          title: 'Foo Bar',
-          url:   'http://example.com',
-          language: '/rest/v1/language/100/'
-        }
+        page_id: page.id,
+        name: "this-page-does-not-exist-13172404",
+        title: 'Foo Bar',
+        url:   'http://example.com',
+        language: '/rest/v1/language/100/'
+      }
```
Comment (NealJMD) [inline spec/requests/pages_spec.rb]: y u remove indent?

File: spec/requests/pages_spec.rb [hunk 5]
```diff
@@ -56,17 +101,17 @@
       it 'creats a petitionform' do
```
Comment (NealJMD) [inline spec/requests/pages_spec.rb]: typo

File: spec/requests/pages_spec.rb [hunk 6]
```diff
@@ -56,17 +101,17 @@
...
+            {
+              :client_hosted=>true,
+              :client_url=>"http://example.com",
+              :ask_text=>"Dummy ask",
+              :thank_you_text=>"Dummy thank you",
+              :statement_text=>"Dummy statement",
+              :page=>"https://act.sumofus.org/rest/v1/petitionpage/12574/"
+            }
           )
+              )
```
Comment (NealJMD) [inline spec/requests/pages_spec.rb]: I don't understand - it looks like you removed a bunch of indentation but it makes the formatting look kinda worse.
```
            }
          )
              )
        )
```
is pretty wonky looking

**Comment by NealJMD** [comment]
the part where you unindented a bunch of stuff I don't understand at all, but the parts you actually changed look :spades: :spades: to me

File: app/workers/queue_listener.rb (hunk 1 above)
Comment (osahyoun) [inline app/workers/queue_listener.rb]: the only required param is the page id. we can't know what's going to be sent to the worker.

File: app/workers/queue_listener.rb (hunk 2 above)
Comment (osahyoun) [inline app/workers/queue_listener.rb]: the regular expression has brackets. %r means I don't have to escape them.

File: app/workers/queue_listener.rb (hunk 1 above)
Comment (NealJMD) [inline app/workers/queue_listener.rb]: Looking at [this part](https://roboticdogs.actionkit.com/docs/manual/api/pages.html) of the AK docs, it says 

> Only the fields name and title are required. 

Am I missing something?

File: app/workers/queue_listener.rb (hunk 2 above)
Comment (NealJMD) [inline app/workers/queue_listener.rb]: ahh nice one

File: spec/requests/pages_spec.rb (hunk 3 above)
Comment (osahyoun) [inline spec/requests/pages_spec.rb]: good, but does that me my specs are typically unintelligible?

File: spec/requests/pages_spec.rb (hunk 4 above)
Comment (osahyoun) [inline spec/requests/pages_spec.rb]: oh, I don't know why?

File: spec/requests/pages_spec.rb (hunk 3 above)
Comment (NealJMD) [inline spec/requests/pages_spec.rb]: hahah no just some positive code review

File: spec/requests/pages_spec.rb (hunk 6 above)
Comment (osahyoun) [inline spec/requests/pages_spec.rb]: yeah, what's going on there. I'll take a look

**Comment by NealJMD** [comment]
cheers, I'll merge when green

File: app/workers/queue_listener.rb (hunk 1 above)
Comment (osahyoun) [inline app/workers/queue_listener.rb]: I'm updating an existing page, not creating one.

Comment (NealJMD) [inline app/workers/queue_listener.rb]: got it

**Comment by osahyoun** [comment]
Thanks for the review, matey! :tulip:

**Pull request merged**